components/contact/show_inquiry_modal.coffee

```diff
@@ -41,12 +49,37 @@ module.exports = class ShowInquiryModal extends ContactView
     return unless city = @partner.displayLocations @user?.get('location')?.city
     @$('.contact-location').html ", " + city
 
-  submit: (e) ->
+  onSubmit: (e) ->
+    super
+
     analyticsHooks.trigger 'inquiry:show',
       label: modelNameAndIdToLabel 'show', @show.get('id')
+
     @model.set
       inquireable_id: @show.get('id')
       inquireable_type: 'partner_show'
       contact_gallery: true
       session_id: SESSION_ID
-    super
+
+    if sd.FORCED_LOGIN_INQUIRY is 'force_login' and !@user
+      @openInquiryQuestionnaire()
+    else
+      @submit()
+
+  openInquiryQuestionnaire: ->
+    user = User.instantiate _.pick @model.attributes, 'name', 'email'
+
+    fakeArtwork = new ProxyArtwork partner: @partner
```

Comment:
I don't have a better name for `ProxyArtwork`, maybe it should just be a regular `Artwork` with the partner dependency-injected into it? That way the related methods would work.